### PR TITLE
updateSuppressionCrashFix

### PR DESCRIPTION
**Pull request by Kahzerx**

### CODE REVIEW

**Comment by Kahzerx** [comment]
vanilla++

File: src/main/java/carpet/settings/CarpetSettings.java [hunk 1]
```diff
@@ -415,6 +415,9 @@ public String description()
     )
     public static boolean optimizedHardHitBoxEntityCollision = false;
 
+    @Rule(desc = "Fixes updates suppression causing server crashes.", category = BUGFIX)
```
Comment (Fallen-Breath) [inline src/main/java/carpet/settings/CarpetSettings.java]: it's not a bug i think, more like a creative experimental tweak

File: patches/net/minecraft/util/Util.java.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
++import carpet.settings.CarpetSettings;
+ import com.google.common.collect.Iterators;
+ import it.unimi.dsi.fastutil.Hash.Strategy;
+ import java.io.File;
+@@ -158,7 +160,9 @@
+         }
+         catch (ExecutionException executionexception)
+         {
+-            logger.fatal("Error executing task", (Throwable)executionexception);
++            if(!CarpetSettings.updateSuppressionCrashFix || !(executionexception.getCause() instanceof ThrowableSuppression)) {
```
Comment (Fallen-Breath) [inline patches/net/minecraft/util/Util.java.patch]: maybe leaving a neat message to the console to indicate someone is update suppressing will be nicer

File: src/main/java/carpet/helpers/ThrowableSuppression.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package carpet.helpers;
+
+public class ThrowableSuppression extends RuntimeException {
```
Comment (Fallen-Breath) [inline src/main/java/carpet/helpers/ThrowableSuppression.java]: will it be nicer if you use aname like `UpdateSuppressionException` to show the usage more accurately

**Pull request merged**